src/components/Payment.js

```diff
@@ -63,6 +77,12 @@ class Payment extends Component {
             )
           ),
           this.renderAmountField(false)('Tip in Credit')
+        ),
+        CloseOrderBtn(
+          {
+            shouldBeDisabled: this.state.balance !== 0,
+            handleClick: this.props.closeOrder.bind.apply(this.props.closeOrder, [null].concat(this.allInputAmounts()))
```

Comment:
Kinda iffy on this whole thing.. regarding both passing the bound callback to child component, and the retrieval of payment / tip amounts (`allInputAmounts`).